sklearn/calibration.py

```diff
@@ -175,7 +175,8 @@ def fit(self, X, y, sample_weight=None):
                     this_estimator.fit(X[train], y[train])
 
                 calibrated_classifier = _CalibratedClassifier(
-                    this_estimator, method=self.method)
+                    this_estimator, method=self.method,
+                    classes=np.unique(y[train]))
```

Comment:
This is related to my concern that "It's also unclear to me that the current implementation correctly handles the case that classes are absent for some training sets."

Could you concoct a test case, with toy data and your own train-test splits, say, that checks this issue?